react-app/src/components/Common/clinkmenu.js

```diff
@@ -136,32 +157,60 @@ const ClinkMenu = (props) => {
           }}
           form={shareForm}
         >
-          <Form.Item label="Share write access by username" name="username"
+          <Form.Item label="Share write access by username" name="toShare"
             rules={[
               {
-                required: false,
+                required: true,
               },
             ]}
           >
             <Select mode="multiple" 
               filterOption={(input, option) =>
                 option.children.toLowerCase().includes(input.toLowerCase())
               }>
-              {users.map(user => (
+              {usersToShare.map(user => (
                 <Option value={user.id}>{user.username}</Option>
               ))}
             </Select>
           </Form.Item>
-          <Form.Item label="Share read access by link" name="link"
+          <Form.Item>
+            <Button type="primary" htmlType="submit" loading={isLoading}>
+              Submit
+            </Button>
+          </Form.Item>
+        </Form>
+        <Form {...layout} name="unshare-clink" 
+        onFinish={onUnshareFinish} 
+        onFinishFailed={onFinishFailed}
+          initialValues={{
+            remember: false,
+          }}
+          form={unshareForm}
+        > 
+          <Form.Item label="Remove write access:" name="toRemove"
             rules={[
               {
-                required: false,
+                required: true,
               },
             ]}
           >
-            {/* clink url goes here */}
+            <Select mode="multiple" 
+              filterOption={(input, option) =>
+                option.children.toLowerCase().indexOf(input.toLowerCase()) >= 0
+              }>
+              {sharedUsers.map(user => (
+                <Option value={user.id}>{user.username}</Option>
+              ))}
```

Comment:
should we exclude ourselves as an option to remove write access from? if not, when I remove write access for myself on a clink that I own, will the clink not display in the sidebar/user page?

Reply:
Yeah I was wondering. but if you were trying to remove yourself, how would you do it?
UX/UI wise there is probably a better way to do this, but we're a week out from launch so I'd leave it and try to get as much of the other functionality up as possible.
Unless we don't want the ability to remove yourself at all? Only other people can remove you?